File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution(object):
    def longestMonotonicSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        inc,de,maxc=1,1,1
        for  i in range(len(nums)-1):
            if(nums[i]<nums[i+1]):
                inc+=1
                de=1
            elif(nums[i]>nums[i+1]):
                de+=1
                inc=1
            else:
                inc,de=1,1
            maxc=max(maxc,max(inc,de))
        return maxc
        
    
        

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int increaseCount=1,decreaseCount=1,maxlen=1;
        int n=nums.length;
        for(int i=0;i<n-1;i++){
            if(nums[i]<nums[i+1]){
                increaseCount++;
                decreaseCount=1;

            }
            else if(nums[i]>nums[i+1]){
                decreaseCount++;
                increaseCount=1;
            }
            else{
                increaseCount=1;
                decreaseCount=1;
            }
        maxlen=Math.max(maxlen,Math.max(increaseCount,decreaseCount));
        }
        return maxlen;
        
        
    }
}